api/v1alpha1/lua_types.go

```diff
@@ -0,0 +1,41 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import gwapiv1 "sigs.k8s.io/gateway-api/apis/v1"
+
+// Lua defines a Lua extension
+// Only one of Source or Disabled must be set
+//
+// +kubebuilder:validation:XValidation:rule="has(self.source) ? (!has(self.disabled) || self.disabled == false) : (has(self.disabled) && self.disabled == true)",message="Exactly one of source or disabled must be set."
+type Lua struct {
+	// Source is the LuaSource that will be executed for this extension.
+	//
+	// +optional
+	Source *LuaSource `json:"source"`
+	// Disabled is the status of the extension.
+	// Used for disabling Lua extensions at a route level.
+	//
+	// +optional
+	Disabled *bool `json:"disabled,omitempty"`
+}
+
+// LuaSource contains source code information for a user defined Lua script
+// Only one of Inline or ValueRef must be set
+//
+// +kubebuilder:validation:XValidation:rule="has(self.inline) ? !has(self.valueRef) : has(self.valueRef)",message="Exactly one of inline or valueRef must be set."
+type LuaSource struct {
+	// Inline contains the value as an inline string.
+	//
+	// +optional
+	Inline *string `json:"inline,omitempty"`
+	// ValueRef contains the contents of the body
+	// specified as a local object reference.
+	// Only a reference to ConfigMap is supported.
```

Comment:
need cel validation for this `Only a reference to ConfigMap is supported.`

what about the configmap key?

Reply:
Added checks for object kind and default core group.

Reply:
Also added a mention of the key `lua` that will be used for fetching the value.